pkg/providers/snow/apibuilder.go

```diff
@@ -63,6 +63,8 @@ func KubeadmControlPlane(log logr.Logger, clusterSpec *cluster.Spec, snowMachine
 		clusterapi.SetBottlerocketControlContainerImageInKubeadmControlPlane(kcp, clusterSpec.VersionsBundle)
 		clusterapi.SetUnstackedEtcdConfigInKubeadmControlPlaneForBottlerocket(kcp, clusterSpec.Cluster.Spec.ExternalEtcdConfiguration)
 		addBottlerocketBootstrapSnowInKubeadmControlPlane(kcp, clusterSpec.VersionsBundle.Snow.BottlerocketBootstrapSnow)
+		clusterapi.SetBottlerocketAdditionalSettingsInKubeadmControlPlane(kcp, clusterSpec.SnowMachineConfig(
```

Comment:
nit: can we def
```
machineConfig := clusterSpec.SnowMachineConfig(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name)
```

then use it for multiple places, same for kct, etcdcluster etc.